North-Seattle-College/ad440-winter2022-thursday-repo
# Deploy emotion model

closes #157 
Train and deploy emotion model using Sagemaker, and built-in algo "blazingtext."

** I suggest not running each step the notebooks since these process a lot of data and training costs..

Updated Emotions2.ipynb steps and added datacleaning:
1. Get Data: Get feedback data from s3 bucket
2. Clean dataset: remove stopwords, punctuation,  numbers and convert to lower case
3. Find emotion per sentence
* We are using the text2emotion library for determining the emotion per sentence by matching certain words with emotion.
4. Add emotion to each sentence
5. Export analysis to CSV for training and deploying the model

Created emotions-model.ipynb for training and deployment for endpoint
1. Get data from csv created in Emotions2.ipynb
2. Convert emotions to numerical value
3. Prepare dataset: 
* move emotional value to first column
* add __label___ for the supervised learning
* tokenize sentences
4. split data into 80% training 20% validation
5. Upload training and validation csv files to notebook and s3 bucket.
6. Train the model:
* set up estimator with parameters to launch the training job
7. Train the algorithm
8. Get train accuracy and validation accuracy
9. Deploy the model / endpoint - test sentences and check to see if the emotional value is correct.
10. delete the endpoint to save costs.

TIME: 20 hours?!

**TEST** 
1.  Go to sagemaker console
2. Click notebook in left navigation bar
3. Choose notebook instances
4. find emotion-analysis notebook
5. find [emotion-analysis] > choose "open jupyter" https://emotion-analysis-nvi9.notebook.us-west-2.sagemaker.aws/tree
6. open emotions2 file
7. review steps (i suggest not running everything)
8. open emotions-deploy
9. review steps
10. if you want to test the endpoint start on 18 (under hosting) and add your own sentence. it will take a few minutes to create a new endpoint. Make sure you delete the endpoint afterwards.
12. All training jobs are in s3 bucket: 
<img width="1199" alt="s3 bucket" src="https://user-images.githubusercontent.com/49288075/160024627-ab4445a2-9099-45ca-81ba-6084238f4ad8.png">
<img width="735" alt="traindeploy" src="https://user-images.githubusercontent.com/49288075/160024746-963b36d6-9b69-4bf5-80bc-576bb829b1cf.png">
<img width="705" alt="response" src="https://user-images.githubusercontent.com/49288075/160024924-2d3287ef-c106-48ec-8163-f4e7acf95f61.png">
